llvm/lib/Target/AIE/AIEWawRegRewriter.cpp

```diff
@@ -82,16 +85,25 @@ class AIEWawRegRewriter : public MachineFunctionPass {
   LiveRegMatrix *LRM = nullptr;
   LiveIntervals *LIS = nullptr;
   const TargetInstrInfo *TII = nullptr;
+  /// Vector to keep track which additional registers are blocked
+  /// by
+  /// a) being unused GPR registers within the MachineFunction or \
+  /// b) being Callee save registers. \
+  /// Unused GPR registers can be used for spill destinations, therefore
+  /// avoiding costly spills to memory.
+  PhysRegVec AdditionalBlockedRegs;
 
   bool renameMBBPhysRegs(const MachineBasicBlock *MBB);
 
-  /// Get all the defined physical registers that the MachineBasicBlock already
-  /// uses. These physical registers should not be used for replacement
-  /// candidates, since this would introduce new WAW dependencies, which this
-  /// pass tries to remove.
-  BitVector getDefinedPhysRegs(const MachineBasicBlock *MBB) const;
+  /// Return a Bitvector with all the defined physical registers of the
+  /// \param MBB. These physical registers should not be
+  /// used for replacement candidates, since this would introduce new WAW
+  /// dependencies, which this pass tries to remove. Additionally, block all the
+  /// physical register in \param AdditionalBlockedRegs.
+  BitVector getBlockedPhysRegs(const MachineBasicBlock *MBB,
```

Comment:
We are passing parameter that is also a class member. Why not access it directly?